### PR TITLE
Remove the dependency between the orchestrator container and the worker app

### PR DESCRIPTION
**Pull request by benmoriceau**
## What
Remove the dependency from the orchestrator app to the worker app.

It is done by adding a new module `airbyte-commons-worker`. This module is containing the classes needed by the `airbyte-container-orchestrator` to the common module.

## Recommended reading order
1. `build.gradle` of `airbyte-container-orchestrator` -> remove the worker dependencies
2. `airbyte-commons-worker` that contains a set of files that are move from the worker module to the commons worker module.

### CODE REVIEW

**Comment by benmoriceau** [comment]
@colesnodgrass @alovew The build should pass so it is ready to review.

File: airbyte-commons-worker/build.gradle [hunk 1]
```diff
@@ -0,0 +1,65 @@
+import org.jsonschema2pojo.SourceType
+
+plugins {
+    id "java-library"
+    id 'com.github.eirnym.js2p' version '1.0'
+}
+
+group 'io.airbyte...'
+version '0.40.14'
```
Comment (jdpgrailsdev) [inline airbyte-commons-worker/build.gradle:9]: I think that `group` and `version` can be removed from this file.  This is handled by the `allprojects` block in the root `build.gradle` file.

File: airbyte-commons-worker/build.gradle [hunk 2]
```diff
@@ -0,0 +1,65 @@
...
+
+repositories {
```
Comment (jdpgrailsdev) [inline airbyte-commons-worker/build.gradle:11]: This whole block can be removed -- it is added by the `subprojects` block of the root `build.gradle` file.

File: airbyte-commons-worker/build.gradle [hunk 3]
```diff
@@ -0,0 +1,65 @@
...
+    targetPackage = 'io.airbyte.persistence.job.models'
+
+    useLongIntegers = true
+    generateBuilders = true
+    includeConstructors = false
+    includeSetters = true
+}
+
+Properties env = new Properties()
+rootProject.file('.env.dev').withInputStream { env.load(it) }
```
Comment (jdpgrailsdev) [inline airbyte-commons-worker/build.gradle:63]: These two lines can be removed.  They are only necessary if you need to load environment variables from the .env file for use in a task (e.g. run, tests, etc).  They don't appear to be used anywhere in this project.

File: airbyte-commons-worker/build.gradle [hunk 4]
```diff
@@ -0,0 +1,65 @@
...
+    testImplementation project(':airbyte-commons-docker')
+}
+
+jsonSchema2Pojo {
+    sourceType = SourceType.YAMLSCHEMA
+    source = files("${sourceSets.main.output.resourcesDir}/workers_models")
+    targetDirectory = new File(project.buildDir, 'generated/src/gen/java/')
+    removeOldOutput = true
+
+    targetPackage = 'io.airbyte.persistence.job.models'
```
Comment (jdpgrailsdev) [inline airbyte-commons-worker/build.gradle:54]: Is this the correct target package for these generated classes?

File: airbyte-commons-worker/build.gradle (hunk 2 above)
Comment (benmoriceau) [reply airbyte-commons-worker/build.gradle:11]: Sorry it was an auto generate file, I removed the group and airbyte version as well.

File: airbyte-commons-worker/build.gradle (hunk 3 above)
Comment (benmoriceau) [reply airbyte-commons-worker/build.gradle:63]: Ok, I had issue with `S3DocumentStoreClientTest` which seemed to miss an env variable. Will retry without.

File: airbyte-commons-worker/build.gradle (hunk 4 above)
Comment (benmoriceau) [reply airbyte-commons-worker/build.gradle:54]: I wanted to limit the change of the PR so the package name are the same than the ones in the worker module. I'll included it in https://github.com/airbytehq/airbyte/pull/17809. Does it sounds ok to you?

**Review by colesnodgrass — APPROVED** [review]

**Review by jdpgrailsdev — APPROVED** [review]
:shipit:

**Comment by benmoriceau** [comment]
@colesnodgrass @jdpgrailsdev I have added https://github.com/airbytehq/airbyte/pull/17570/commits/628b7215f37004e0e0a5e04a4cc6926f6bdb724f to fix the acceptance test.

**Review by jdpgrailsdev — APPROVED** [review]
:shipit:

**Pull request merged**